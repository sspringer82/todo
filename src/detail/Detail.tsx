import React from "react";
import { useParams } from "react-router";
import Done from '../done/Done';
import Subtask from '../Subtask/Subtask';
import { useTodo } from "../TodoContext";

const Detail: React.FC = () => {
  const [todos] = useTodo();
  const { id } = useParams<{ id: string }>();

  const todo = todos.find((todo) => todo.id === parseInt(id, 10));

  return (
    <div>
      { todo && <Done todo={todo} /> }
      <div>{todo?.title}</div>
      <div>{todo?.comment}</div>
      <Subtask subtasks={todo?.subtask} />
    </div>
  );
};

export default Detail;
