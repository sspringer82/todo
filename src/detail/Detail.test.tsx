import { render, screen } from "@testing-library/react";
import Detail, { Props } from "./Detail";

describe("Detail", () => {
  let props: Props;

  beforeEach(() => {
    props = {
      todo: { id: 1, title: "New Todo", done: true, comment: "Todo Comment" },
      onSave: jest.fn(),
    };
  });

  it("should render details of a todo", async () => {
    const {getByTestId} = render(<Detail {...props} />);
    const title = getByTestId('detail-title');
    const comment = getByTestId('detail-comment');
    const done = getByTestId('done-done');
    expect(title).toHaveTextContent('New Todo');
    expect(comment).toHaveTextContent('Todo Comment');
    expect(done).toBeInTheDocument();
  });
  it("should render details of a todo with subtasks", () => {

  });
  it("should be possible to change the state of the todo", () => {});
});
