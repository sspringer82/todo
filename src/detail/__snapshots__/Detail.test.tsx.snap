// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail should perform a snapshot test 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <div
      class="flex"
    >
      <div>
        <button
          class="focus:outline-none relative"
          data-testid="done-button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            data-testid="done-done"
            focusable="false"
            style="color: limegreen;"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </button>
      </div>
      <div
        data-testid="detail-title"
      >
        New Todo
      </div>
    </div>
    <div
      data-testid="detail-comment"
      style="max-width: 422px;"
    >
      Todo Comment
    </div>
    <div
      class="flex mb-2"
      style="margin: 10px -10px;"
    >
      <div
        style="height: 5px; width: 5px; background-color: rgb(52, 211, 153); margin-top: 18px;"
      />
      <div
        style="height: 0px; border-top: 1px solid #34D399; flex-grow: 1; margin-top: 18px;"
      />
      <div
        style="margin-top: 11px; margin-left: 7px; margin-right: 10px; height: 8px; width: 16px; background-color: rgb(52, 211, 153); transform: skew(-70deg);"
      />
      <div
        style="height: 0px; border-top: 1px solid #34D399; flex-grow: 1; margin-top: 11px;"
      />
      <div
        class="mx-2"
        style="color: rgb(52, 211, 153);"
      >
        Subtasks
         
      </div>
      <div
        style="height: 0px; border-top: 1px solid #34D399; flex-grow: 1; margin-top: 11px;"
      />
      <div
        style="margin-top: 11px; margin-left: 7px; margin-right: 10px; height: 8px; width: 16px; background-color: rgb(52, 211, 153); transform: skew(70deg);"
      />
      <div
        style="height: 0px; border-top: 1px solid #34D399; flex-grow: 1; margin-top: 18px;"
      />
      <div
        style="height: 5px; width: 5px; background-color: rgb(52, 211, 153); margin-top: 18px;"
      />
    </div>
    <div
      data-testid="detail-nosubtasks"
    >
      Keine Unteraufgaben
    </div>
    <div>
      Created: 
    </div>
    <div>
      Updated: 
    </div>
    <div>
      <button
        class="px-5 py-1 bg-opacity-25 focus:outline-none border"
        style="border-color: #34d399;"
      >
        close
      </button>
    </div>
  </div>
</div>
`;
